lib/Models/Catalog/Esri/ArcGisMapServerCatalogItem.ts

```diff
@@ -452,9 +475,52 @@ export default class ArcGisMapServerCatalogItem extends UrlMixin(
         return;
       }
 
+      function windowDurationInMs(
+        windowDuration: number | undefined,
+        timeUnit: string | undefined
+      ): number | undefined {
+        if (
+          windowDuration === undefined ||
+          windowDuration === 0 ||
+          timeUnit === undefined
+        ) {
+          return undefined;
+        }
```

Comment:
Just a nitpick. I think we can avoid checks like these, if we make `TimeParams` stricter like:

```ts
interface TimeParams {
  currentTime: number;
  timeWindowDuration: number;
  timeWindowUnit: number;
}```

and make `getTimeParams()` return `getTimeParams(): TimeParams | undefined` if any one of the field is invalid.

Reply:
Can't do that because we need to support the case where only `currentTime` is defined.